src/LogMonitorEntry.js

```diff
@@ -31,6 +31,7 @@ export default class LogMonitorEntry extends Component {
   constructor(props) {
     super(props);
     this.handleActionClick = this.handleActionClick.bind(this);
+    this.previousState = this.props.previousState || this.props.state;
```

Comment:
This is incorrect. Component can receive new props in `componentWillReceiveProps` so it is wrong to calculate anything based on props in the constructor and cache it.